src/api/ITooltip.js

```diff
@@ -74,6 +82,13 @@
     ITooltip.prototype.tooltipUpdate = function () {
         var classed = this.tooltip.attr("class");
         classed = classed.split(" notick").join("") + (this.tooltipTick() ? "" : " notick") + (this.tooltipStyle() === "none" ? " hidden" : "");
+        classed = classed.split(" ")
+            .filter(function(_class){
+                return _class.indexOf("ITooltip-tooltipStyle-")!==0
+            })
+            .join(" ")
+            ;
+        classed += " ITooltip-tooltipStyle-"+this.tooltipStyle();
```

Comment:
```
.this.tooltip.classed("ITooltip-tooltipStyle-default", this.tooltipstyle()==="default")
.this.tooltip.classed("ITooltip-tooltipStyle-series-table", this.tooltipstyle()==="series-table")
```
Would be cleaner?

Reply:
Unfortunately `this.tooltip` is an object returned from `d3-tip` (not an actual d3 selection). It doesn't offer the `classed` method.